common-service-core/src/main/java/org/zowe/apiml/message/log/ApimlLogger.java

```diff
@@ -67,16 +67,19 @@ public static ApimlLogger empty() {
      * @param parameters for message
      */
     public Message log(String key, Object... parameters) {
-        ObjectUtil.requireNotNull(key, "key can't be null");
-        ObjectUtil.requireNotNull(parameters, "parameters can't be null");
+        Message message;
+        if (key == null) {
+            message = Message.invalidKeyMessage(null);
+        } else {
+            ObjectUtil.requireNotNull(parameters, "parameters can't be null");
 
-        if (messageService != null) {
-            Message message = messageService.createMessage(key, parameters);
-            log(message);
-            return message;
+            if (messageService == null) {
+                return null;
+            }
```

Comment:
This if could be on the top as a quick break

Reply:
We decided to leave this as is. Reason:
When the given key is `null`, we want to return the invalid key message instead of a `null` message.